google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/BigQueryConfig.scala

```diff
@@ -0,0 +1,191 @@
+    apply(host, port, credentials)
+
+}
+
+final class ForwardProxy private (val host: String,
+                                  val port: Int,
+                                  val credentials: Option[ForwardProxyCredentials],
+                                  val trustPem: Option[ForwardProxyTrustPem]) {
+
+  /** Java API */
+  def getHost: String = host
+
+  /** Java API */
+  def getPort: Int = port
+
+  /** Java API */
+  def getCredentials: java.util.Optional[ForwardProxyCredentials] = credentials.asJava
+
+  def getForwardProxyTrustPem: java.util.Optional[ForwardProxyTrustPem] = trustPem.asJava
+
+  def withHost(host: String) = copy(host = host)
+  def withPort(port: Int) = copy(port = port)
+  def withCredentials(credentials: ForwardProxyCredentials) = copy(credentials = Option(credentials))
+
+  private def copy(host: String = host,
+                   port: Int = port,
+                   credentials: Option[ForwardProxyCredentials] = credentials,
+                   trustPem: Option[ForwardProxyTrustPem] = trustPem) =
+    new ForwardProxy(host, port, credentials, trustPem)
+
+  override def toString =
+    "ForwardProxy(" +
+    s"host=$host," +
+    s"port=$port," +
+    s"credentials=$credentials," +
+    s"trustPem=$trustPem" +
+    ")"
+
+  override def equals(other: Any): Boolean = other match {
+    case that: ForwardProxy =>
+      Objects.equals(this.host, that.host) &&
+      Objects.equals(this.port, that.port) &&
+      Objects.equals(this.credentials, that.credentials) &&
+      Objects.equals(this.trustPem, that.trustPem)
+    case _ => false
+  }
+
+  override def hashCode(): Int =
+    Objects.hash(host, Int.box(port), credentials)
+}
+
+object BigQueryConfig {
+
+  /** Scala API */
+  def apply(clientEmail: String, privateKey: String, projectId: String, dataset: String)(
+      implicit actorSystem: ActorSystem
+  ): BigQueryConfig = {
+    val session = GoogleSession(clientEmail, privateKey, actorSystem, Option.empty)
+    new BigQueryConfig(projectId, dataset, Option.empty, session)
+  }
+
+  def apply(clientEmail: String, privateKey: String, projectId: String, dataset: String, forwardProxy: ForwardProxy)(
+      implicit actorSystem: ActorSystem
+  ): BigQueryConfig = {
+    val session = GoogleSession(clientEmail, privateKey, actorSystem, Option(forwardProxy))
+    new BigQueryConfig(projectId, dataset, Option(forwardProxy), session)
+  }
+
+  /** Java API */
+  def create(clientEmail: String,
+             privateKey: String,
+             projectId: String,
+             dataset: String,
+             actorSystem: ActorSystem): BigQueryConfig = {
+    apply(clientEmail, privateKey, projectId, dataset)(actorSystem)
+  }
+
+}
+
+final class BigQueryConfig(val projectId: String,
+                           val dataset: String,
+                           val forwardProxy: Option[ForwardProxy],
+                           @InternalApi private[bigquery] val session: GoogleSession) {}
```

Comment:
I think this constructor should be private, `GoogleSession` is a private class so the constructor can't be used.